File: backend/app/schemas/exam_preview.py
import uuid
from decimal import Decimal

from pydantic import BaseModel


class PreviewQuestionOut(BaseModel):
    question_number: int
    prompt_text: str | None
    passage_text: str | None
    is_placeholder: bool


class PreviewBlockOut(BaseModel):
    block_id: uuid.UUID
    section_number: int
    section_label: str
    title: str
    instruction: str | None
    question_start: int | None
    question_end: int | None
    question_count: int
    points: Decimal
    continuation: bool
    questions: list[PreviewQuestionOut]


class PreviewPageOut(BaseModel):
    page_number: int
    blocks: list[PreviewBlockOut]


class ExamPreviewOut(BaseModel):
    exam_id: uuid.UUID
    title: str
    total_questions: int
    total_points: Decimal
    page_count: int
    pages: list[PreviewPageOut]

File: frontend/src/types/examPreview.ts
export interface PreviewQuestionOut {
  question_number: number;
  prompt_text: string | null;
  passage_text: string | null;
  is_placeholder: boolean;
  part_number: number | null;
  part_title: string | null;
  part_instruction: string | null;
}

export interface PreviewBlockOut {
  block_id: string;
  section_number: number;
  section_label: string;
  title: string;
  instruction: string | null;
  question_start: number | null;
  question_end: number | null;
  question_count: number;
  points: string;
  continuation: boolean;
  questions: PreviewQuestionOut[];
}

export interface PreviewPageOut {
  page_number: number;
  blocks: PreviewBlockOut[];
}

export interface ExamPreviewOut {
  exam_id: string;
  title: string;
  grade_number: number;
  level_code: string;
  total_questions: number;
  total_points: string;
  page_count: number;
  pages: PreviewPageOut[];
}
